tests/Dockerfile

```diff
@@ -1,5 +1,5 @@
 # The Dockerfile's resulting image is purpose-built for executing Jaeger Operator e2e tests within the OpenShift release (https://github.com/openshift/release) using Prow CI.
-FROM golang:1.21
```

Comment:
Why is this change needed?

Reply:
This is an OSSF recommandation. Pining version avoid certain vulnerabilities.